src/docfx/cli/Docfx.cs

```diff
@@ -108,34 +109,25 @@ private static (string command, string workingDirectory, CommandLineOptions opti
-                    syntax.DefineOption("locale", ref options.Locale, "The locale of the docset to build");
                     syntax.DefineOption("o|output", ref options.Output, "Output directory in which to place restore log.");
-                    syntax.DefineOption(
-                        "template", ref options.Template, "The directory or git repository that contains website template.");
-                    syntax.DefineOption("legacy", ref options.Legacy, "Enable legacy output for backward compatibility.");
-                    syntax.DefineOption("v|verbose", ref options.Verbose, "Enable diagnostics console output.");
-                    syntax.DefineParameter("directory", ref workingDirectory, "A directory or subdirectores that contains docfx.yml/docfx.json.");
+                    DefineCommonOptions(syntax, ref workingDirectory, options);
 
                     // Build command
                     syntax.DefineCommand("build", ref command, "Builds a docset.");
                     syntax.DefineOption("o|output", ref options.Output, "Output directory in which to place built artifacts.");
-                    syntax.DefineOption(
-                        "template", ref options.Template, "The directory or git repository that contains website template.");
-                    syntax.DefineOption("legacy", ref options.Legacy, "Enable legacy output for backward compatibility.");
-                    syntax.DefineOption("locale", ref options.Locale, "The locale of the docset to build.");
-                    syntax.DefineOption("v|verbose", ref options.Verbose, "Enable diagnostics console output.");
-                    syntax.DefineParameter("directory", ref workingDirectory, "A directory or subdirectores that contains docfx.yml/docfx.json.");
+                    DefineCommonOptions(syntax, ref workingDirectory, options);
 
                     // Watch command
                     syntax.DefineCommand("watch", ref command, "Previews a docset and watch changes interactively.");
-                    syntax.DefineOption("locale", ref options.Locale, "The locale of the docset to build.");
-                    syntax.DefineOption(
-                        "template", ref options.Template, "The directory or git repository that contains website template.");
                     syntax.DefineOption("port", ref options.Port, "The port of the launched website.");
-                    syntax.DefineOption("v|verbose", ref options.Verbose, "Enable diagnostics console output.");
-                    syntax.DefineParameter("directory", ref workingDirectory, "A directory or subdirectores that contains docfx.yml/docfx.json.");
+                    DefineCommonOptions(syntax, ref workingDirectory, options);
                 });
 
+                if (options.Stdin)
+                {
+                    options.StdinConfig = JsonUtility.Deserialize<JObject>(Console.ReadLine(), new FilePath("--stdin"));
```

Comment:
>--stdin [](start = 109, length = 7)

so the diagnostics will be reported on the file `--stdin`?

Reply:
yep